### PR TITLE
doc: convert coq to commonmark

### LINKED ISSUE #104964: Convert Coq from Docbook to CommonMark
This issue tracks the conversion of `/doc/languages-frameworks/coq.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Instructions for performing this conversion

1. Post in this issue that you intend to work on it. (Please also post if you stop working on it)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/languages-frameworks/index.xml`: change `coq.xml` to `coq.section.xml`.
4. Move `doc/languages-frameworks/coq.xml` to `doc/languages-frameworks/coq.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by wayofthepie**
Converts the coq section in the docs to commonmark, closes #104964. 

I just figured out `pandoc` has a `--wrap none` argument also, which prevents hard-wraps on conversion:

```bash
$ pandoc \
    --from docbook \
    --to commonmark \
    --wrap none \
    languages-frameworks/coq.xml > languages-frameworks/coq.section.md
```

### CODE REVIEW

File: doc/languages-frameworks/index.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
  <xi:include href="android.section.xml" />
  <xi:include href="beam.section.xml" />
  <xi:include href="bower.xml" />
- <xi:include href="coq.xml" />
+ <xi:include href="coq.section.xml" />
```
Comment (SuperSandro2000) [inline doc/languages-frameworks/index.xml:12]: Shouldn't this has a md extension?

**Comment by siraben** [comment]
LGTM. Rendered output looks same as before.

File: doc/languages-frameworks/index.xml (hunk 1 above)
Comment (wayofthepie) [reply doc/languages-frameworks/index.xml:12]: Currently the markdown gets converted back into docbook first and that is what gets rendered to html. This reads that markdown -> docbook converted output.

**Review by ryantm — APPROVED** [review]
Wow, the output is 100% exactly the same!

**Pull request merged**